crates/sound/src/audio_backends/pw_backend.rs

```diff
@@ -210,12 +212,15 @@ impl AudioBackend for PwBackend {
     fn prepare(&self, stream_id: u32) -> Result<()> {
         let mut stream_states = self.stream_states.write().unwrap();
         let mut stream_hash = self.stream_hash.write().unwrap();
+        let mut stream_listener = self.stream_listener.write().unwrap();
         self.thread_loop.lock();
         let stream_params = self.stream_params.read().unwrap();
         let params = stream_params[stream_id as usize].clone();
         //dbg!("params: {}", params);
 
-        if stream_states[stream_id as usize] == VIRTIO_SND_R_PCM_SET_PARAMS {
+        if stream_states[stream_id as usize] == VIRTIO_SND_R_PCM_SET_PARAMS
+            || stream_states[stream_id as usize] == VIRTIO_SND_R_PCM_PREPARE
```

Comment:
I think if guest sends the PREPARE cmd and the stream is in the PREPARE state, this would end up creating the stream again. I think we need to handle the case in which guest sends SET_PARAM and the stream is in the prepare state. In that case, we have to destroy the stream (maybe call release?) and moves the state to SET_PARAM. The next call to PREPARE would create the stream again.